### ISSUE TITLE
Seeking community feedback for Import/Export Data-Tier Application Framework Wizard extension

### ISSUE THREAD

**Issue by yualan** [microsoft/azuredatastudio#3171]
As part of bringing over key features from SSMS, the team is invested in bringing over DacFx support in Azure Data Studio, starting with the Export and Import bacpac wizards in SSMS. According to telemetry, these are the second most used wizards in SSMS after the Import Flat File Wizard. 

By bringing over these experiences, we have an opportunity to improve the existing experience so that users have a better idea of knowing when to use Extract/Export/Import or more guidance for when to use DACPACs or BACPACs.

Here are some GIFs of the current experience.

**Import**
![importbacpac](https://user-images.githubusercontent.com/30873802/48224627-38d1df80-e34f-11e8-805c-ff9dc48762f0.gif)

**Export**
![exportbacpac](https://user-images.githubusercontent.com/30873802/48224640-4a1aec00-e34f-11e8-9266-6ed4f9eb83a3.gif)

We would appreciate your help by letting us know your thoughts of the current experience, and suggestions you have in mind as we continue to build out this feature.

You can keep track of current work here: https://github.com/Microsoft/azuredatastudio/issues/3163

**Comment by joshbooker** [comment]
@yualan Thanks for this.  Can you share how to test this.  I tried cloning and building `feature/dacfxexport` but the export bacpac did not succeed.  Upon clicking Export button there was no task progress displayed and no bacpac file was created.

**Comment by joshbooker** [comment]
Bug:  I found that when I change the bacpac location, the default location still shows on the last page of the wizard.
Suggestion: it would be nice if Import\Export location defaulted to the current root path open in the file explorer.

Thanks for listening.

**Comment by joshbooker** [comment]
Are there plans to include the Schema Compare like [DacFx Script action](https://blogs.msdn.microsoft.com/ssdt/2016/09/20/schema-compare-in-sqlpackage-and-the-data-tier-application-framework-dacfx/)?  If so, how about including a way to execute in [sql command mode](https://docs.microsoft.com/en-us/sql/tools/sqlcmd-utility?view=sql-server-2017)

Thanks for listening?

**Comment by joshbooker** [comment]
@yualan , I'm Looking forward to providing feedback to improve the cmd line experience for Extract, Compare, Update

[This](https://github.com/joshbooker/sqlOS-Scripter-Extract-Compare-Update) is what we're working with now.  

![alt demo](https://joshbooker.github.io/DEMO5_compareUpdate.gif)

Would love to see this become a wizard experience in ADS.

**Comment by joshbooker** [comment]
**Question:** it appears the DacFx Script action is not included here: Microsoft/sqltoolsservice#728 
Any chance of adding the script action before this merges?

Scripting changes rather than publishing directly to database is really a important scenario...essential for maintaining the DEV to DBA relationship. :-)

Thanks for listening.

**Comment by armordba** [comment]
Hi Alan,
    I currently use SSDT via Visual Studio with Azure Dev Ops for
our source control and project planning.  We have continuous integration
continuous deployment.  I have a MAC and I speak around the country.  I
would love to use a product which would support SSDT on a MAC, so I am
being to explore Azure Data Studio to see if it offers which I need to use
my Mac without having to use a VM with windows.

Thanks,

Tom

**Comment by yualan** [comment]
Hi @armordba , what are the main features you are looking for from SSDT?

**Comment by armordba** [comment]
1. SQL Object creation and verification, for example, when creating a view
or stored procedure, SSDT will verify the table structure it is working
with and not allow the project to build.
2. If the build is successful, you can then publish the database project to
a database or create one if the database is not there.
3. Pre and Post deployment scripts I can add to the system.
4. Database options can be set and published.
5. Ability to check the database scripts into Source Control.

**Comment by joshbooker** [comment]
@yualan & @kisantia 

It appears the DacFx Script action is not included here: Microsoft/sqltoolsservice#728

Question: Any chance of adding the script action before this merges?

Scripting changes rather than publishing directly to database is really a important scenario...essential for maintaining the DEV to DBA relationship. :-)

Thanks for listening.

cc: https://github.com/Microsoft/azuredatastudio/issues/3163#issuecomment-437585855

**Comment by kisantia** [comment]
@joshbooker I will look into adding the Generate Script action to the deploy scenario, but it won't make it into the December release. 

Thanks,
Kim

**Comment by joshbooker** [comment]
@kisantia thanks so much for your reply. Looking forward to it. Have a great day!

**Comment by Alain1405** [comment]
So I'm trying to import an Azure SQL backup in bacpac format into my local SQL Server on my Mac.
I've installed the `import` extension yet the Data-tier application wizard is not available, only the Flat file import. 
Am I missing something?

<img width="1049" alt="screen shot 2019-01-04 at 3 48 21 pm" src="https://user-images.githubusercontent.com/2998545/50693837-7776d400-1038-11e9-823e-e6418ab3e21a.png">

**Comment by Alain1405** [comment]
Actually, I was supposed to use an insider build. Using`1.3.6-insider` build, however, I encounter another issue, the `Next` button stays offuscated once I try to import a `bacpac` file.

**Comment by Alain1405** [comment]
I've downloaded the source for version `1.4.0` and built on Mac, to find that I had the same issue of `Next` button being disabled. I then opened `Chrome Developer Tools` and manually removed the CSS `disabled` configurations. Once the button was enabled, the import worked out.

**Comment by kisantia** [comment]
@Alain1405 thanks for finding this bug! I checked in the fix #3691 and it will be included in the next release of ADS. 

A workaround in the current version is to click the `Previous` button to go back to the Step 1 page, then click the `Next` button to go back to the 'Select Import Bacpac Settings' page and the `Next` button should be enabled if the required fields are filled in.

**Comment by mikefayer** [comment]
I'm working on Ubuntu 18.04, trying to import a bacpac file. I had the same problem as @Alain1405, where the next button is stuck disabled. Your workaround fixed it for me (hit previous, then next), but when I hit "Import" on the summary screen, the pane immediately disappears and nothing happens. There's nothing on any of the logs.

**Comment by kisantia** [comment]
@mikefayer which version of ADS are you using? The latest release 1.3.9 has the fix for this and was working when I tested this release on my Ubuntu 16.04 VM.

**Comment by SQLvariant** [comment]
@yualan I have tons of feedback for you on this topic!  One piece of that feedback is: since ADS is multi-lingual, we want to be able to export a .DACPAC of all our databases with just 2 lines of code.  Have a look at this old request: https://feedback.azure.com/forums/908035-sql-server/suggestions/32902240-sqlpackage-exe-needs-to-be-made-into-at-least-3

I also have a lot more feedback for you, beyond just the cmdline abilities.  I will read through this page and then comment more.

**Comment by mikefayer** [comment]
@kisantia Also works for me with the latest verison, thanks.

**Comment by joshbooker** [comment]
@yualan  and @kisantia  Are there any tracking issues for feature request related to this one including:

[SqlCmd](https://docs.microsoft.com/en-us/sql/tools/sqlcmd-utility?view=sql-server-2017) integration and  [ms-sqlscripter](https://github.com/Microsoft/mssql-scripter) integration

**Comment by paschott** [comment]
Tried to export a bacpac from one of our servers with the build released today. It threw an error. I can't tell for sure what the error is because I can't find a way to read it, but I'd _suspect_ that it has something to do with the fact that there's a SQL user tied to a SQL Login in the database.

Shouldn't we have a way to _not_ script those out as part of creating a bacpac?  I shouldn't have to drop users/logins just to create a bacpac file to use elsewhere.

So two issues:
1. Can't read the error message.
2. Creating a bacpac shouldn't try to script out users tied to logins when we already know this will throw an error. (or at least give an option to not script users)

**Comment by kisantia** [comment]
@paschott can you try exporting using [sqlpackage ](https://docs.microsoft.com/en-us/sql/tools/sqlpackage-download?view=sql-server-2017)and email me the log file? My email is kisantia@microsoft.com

Example export command:
sqlpackage.exe /Action:Export /ssn:[server].database.windows.net /sdn:[database] /su:[user] /sp:[password] /tf:"C:\Temp\database.bacpac" /df:sqlpackage-export.log

**Comment by ffcruz85** [comment]
regarding what @joshbooker said, I think it would be nice if the path to save box showed last pah used and/or add a dropdown for recent folders used

**Comment by ffcruz85** [comment]
and another one that would be an awesome feature for me: select multiple DBs at once ... really annoying to go through the wizard 20 times...
Since we can't multiple select DBs to right click in the connections pane, maybe it would have to be an option within the wizard?
Either way, this would be great

**Comment by kunal-mandalia** [comment]
Just wanted to say this is a great extension @yualan ❤️ 

It'd be nice if in the ADS extension search the term "import" would also bring this up. I only learned about the extension through this github issue.

**Comment by paschott** [comment]
Tried again today - had a Windows login tied to a user and I know that's not allowed in a bacpac, but it would be _great_ to have an option to ignore those logins/users when generating a bacpac. I've been blocked by so many small things when trying to generate dacpacs/bacpacs using the GUI tools that it's been quite frustrating trying to move databases around this way.

**Comment by kburtram** [comment]
Closing older issues.  Thanks for all the great feedback that we used during feature planning!
